### PR TITLE
[R-120-17] Refactoring to reduce pain in debugging and REPL work

### PR DESCRIPTION
**Pull request by parsonsmatt**
## Description

Oh man, I got a little wild here.

- Banished a ton of redundant constraints and removed them entirely from the acid-state query interface. Now you can load the database without needing to conjure the entire app stack!
- Liberal use of `-fwarn-redundant-constraints` rooted out a metric ton of them, and now we have many functions that are much easier to call thanks to fewer constraints.
- Uh, I'll fix the actual bug soon, now that I can test this out.

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: wallet/cardano-sl-wallet.cabal [hunk 1]
```diff
@@ -181,6 +181,7 @@ library
   ghc-options:         -Wall
                        -fno-warn-orphans
                        -O2
+                       -fwarn-redundant-constraints
```
Comment (erikd) [inline wallet/cardano-sl-wallet.cabal]: Didn't know about that. I suspect that will be useful!

Comment (erikd) [reply wallet/cardano-sl-wallet.cabal]: Actually, I think that flag was part of `-Wall` in ghc 8.0.1 but got removed in 8.0.2

Comment (cleverca22) [reply wallet/cardano-sl-wallet.cabal]: i tend to use `-Weverything`, even more then all!

Comment (parsonsmatt) [reply wallet/cardano-sl-wallet.cabal]: `-Wno-really-everything`

**Comment by cleverca22** [comment]
```
[20 of 30] Compiling Pos.Explorer.Web.ClientTypes ( src/Pos/Explorer/Web/ClientTypes.hs, dist/build/Pos/Explorer/Web/ClientTypes.o )
  |  
  | src/Pos/Explorer/Web/ClientTypes.hs:116:55: error:
  | • Overlapping instances for NFData CHash
  | arising from the 'deriving' clause of a data type declaration
```

**Pull request merged**